### DISCUSSION TITLE
Unable to install block2 on conda environment

### DISCUSSION THREAD

**Discussion by JessicaJohnBritto** [block-hczhai/block2-preview#116] [Q&A] [answered]
I am using anaconda's default python interpreter. I created a new virtual environment (AJJ) on conda and installed latest version of numpy, matplotlib, scipy using `conda install <package-name>`. And the pip version in this environment is `24.0` and I did not install this separately.

To install block2, on using this command `python -m pip install block2` and also with this `pip install block2`, I am getting this error in both the cases - `ERROR: Could not build wheels for block2, which is required to install pyproject.toml-based projects`.

Then, I tried this command `python -m pip install --no-binary :all: block2` which took a lot of time and in the end, it gave me this error `ERROR: Cannot install block2==0.1.10, block2==0.1.4, block2==0.1.5, block2==0.1.6, block2==0.1.7, block2==0.1.8, block2==0.2.0, block2==0.3.0, block2==0.4.0, block2==0.4.1, block2==0.4.10, block2==0.4.12, block2==0.4.13, block2==0.4.14, block2==0.4.2, block2==0.4.5, block2==0.4.6, block2==0.4.8, block2==0.4.9, block2==0.5.0, block2==0.5.1 and block2==0.5.2 because these package versions have conflicting dependencies.`

![image](https://github.com/user-attachments/assets/b679b081-6ff9-4ddd-941f-a479197d8478)

I am looking for a method so that I can install this software package that is anaconda friendly. Any help will be appreciated.

Thanks

**Comment by hczhai** [comment]
What is your Python version and what is the operating system?

**Reply by JessicaJohnBritto** [reply]
Python version: 3.12.4, Operating system: Windows

**Reply by hczhai** [reply, marked as answer]
You can install ``block2`` for Python 3.12 and Windows system using the following command:

```
python -m pip install block2==0.5.3rc16 --extra-index-url=https://block-hczhai.github.io/block2-preview/pypi/
```

**Reply by JessicaJohnBritto** [reply]
`pyblock2` is not installed with the above command (could not be found in pip/conda list). It cannot be installed using `pip install pyblock2` and `conda install pyblock2`. 

Errors-
`ERROR: Could not find a version that satisfies the requirement pyblock2 (from versions: none)
ERROR: No matching distribution found for pyblock2`

It is not found in pypi packages either. Unable to run the tutorial notebooks due to this.

`block2` is installed and it's version is `0.5.3rc16`.

**Reply by hczhai** [reply]
You can directly run the tutorial without installing "pyblock2".

**Reply by JessicaJohnBritto** [reply]
I am following the Python Interface Tutorial, and unable to import DMRGDriver and SymmetryTypes `from pyblock2.driver.core import DMRGDriver, SymmetryTypes`. I tried this code with block2 too but it does not seem to have these modules.

Could you also please tell me the difference between block2 and pyblock2?
Thanks

**Reply by hczhai** [reply]
Please attach a **screenshot** showing the output of the following commands (in the same terminal):

First, type (even if you have already done this):

```
python3 -m pip install block2==0.5.3rc16 --extra-index-url=https://block-hczhai.github.io/block2-preview/pypi/
```

And then type (to enter the interpreter):

```
python3
```

And then, inside the Python interpreter, type:

```python3
from pyblock2.driver.core import DMRGDriver, SymmetryTypes
```

**Reply by JessicaJohnBritto** [reply]
Screenshot - 
![image](https://github.com/user-attachments/assets/d31b1841-1542-4391-a735-73d5427f6e91)

**Reply by hczhai** [reply]
Thanks for sharing the information. Please type the following in the terminal:

```
$env:PATH = $env:PATH -split ';' -notmatch '\.' -join ';'
echo $env:PATH
```

And then redo ``python3`` and then in the Python interpreter, try ``from pyblock2.driver.core import DMRGDriver, SymmetryTypes``.

If this does not solve the problem or there are additional error messages, please attach a new screenshot.

**Reply by JessicaJohnBritto** [reply]
I am able to import `pyblock2` when I am in the powershell environment. I have attached screenshot of the same below.
![image](https://github.com/user-attachments/assets/2be54b38-09b4-4f6c-b9ec-930bbdeba213)

When I exit the powershell environment, it gives me the same error while importing `pyblock2`, and the commands shared above only work in the powershell environment.

![image](https://github.com/user-attachments/assets/ad6e3a5c-e4d5-407e-b9e7-fb51b38e8f25)

![image](https://github.com/user-attachments/assets/72b954b5-880d-48bd-ae41-0bab152e02dd)

**Reply by hczhai** [reply]
Thanks for the feedback. The ``$env:PATH = $env:PATH -split ';' -notmatch '\.' -join ';'`` is only supposed to work in PowerShell. If you use Command Prompt, try instead

```
echo %PATH%
set PATH=%PATH:.;=%
echo %PATH%
```

If you want to make this change permanent, you can remove the ``.`` in your PATH environment variable list via system settings, see https://www.mathworks.com/matlabcentral/answers/94933-how-do-i-edit-my-system-path-in-windows .

**Reply by JessicaJohnBritto** [reply]
Yes, this works! Thank you for your help!
